presto-native-execution/presto_cpp/main/PrestoExchangeSource.cpp

```diff
@@ -103,6 +103,8 @@ void PrestoExchangeSource::doRequest() {
                   "Received HTTP {} {}",
                   headers->getStatusCode(),
                   headers->getStatusMessage()));
+        } else if (response->hasError()) {
```

Comment:
Same question

Reply:
The memory allocation for the data response can throw exception: HttpResponse::append which can cause presto process die?